docs/cookbook/using-docker.md

```diff
@@ -0,0 +1,343 @@
+
+NODE_ENV=production
+RUN npm ci
+
+COPY --chown=node . /srv/www/apostrophe/
+
+RUN ./scripts/build-assets.sh
+
+CMD ["node", "app.js"]
+```
+<template v-slot:caption>
+  Dockerfile
+</template>
+
+</AposCodeBlock>
+
+Let's walk briefly through each of the lines. The first line specifies that this image will extend an existing image with the long-term support version of node.js running on an alpine linux v3.15 operating system. If you need another version of node or a newer version of alpine linux, you should alter this line to build from an official node.js [image](https://hub.docker.com/_/node/) or a 3<sup>rd</sup> party image.
+
+The `WORKDIR` command is used to define the working directory of the Docker container where all of the subsequent commands will be run. It implicitly runs both `mkdir` and `cd` commands.
+
+By default, when we issue commands within the `Dockerfile` they are run within the container as a root user. Although Docker should ensure that the 'root' user inside the container can't see or interact with anything outside the container, it never hurts to use a non-root user inside the container too, just in case a flaw in the container system is found. Now that we have a directory, we reassign it to a low-level user `node` using the Docker `RUN` command and the linux `chown` command. Everything following `RUN` will be passed to the command line inside the container. Next, we switch to the new user using the `USER` command.
+
+We are now going to install all of our dependencies inside of the working directory. First, we copy the `package.json` and `package-lock.json` into our project. Next, we pass our `NODE_ENV=production` environment variable into the build and then add those dependencies using `RUN npm ci`.
+
+::: note
+This means that you **must** commit the project `package-lock.json` and you must not list anything required to build the project assets as a "dev" dependency.  Your project doesn't need any of those in production, right? 
+:::
+
+Following the dependency install, all of the necessary project files are copied into the container. Note that we will also create a `.dockerignore` file to exclude some files and folders from being copied.
+
+Next, we run a script to trigger the apostrophe asset build (we will cover this script next). To set a unique `APOS_RELEASE_ID` environment variable each time we change files and redeploy, we are using the script to create a `release-id` file with a unique string. Otherwise, we would have to change this string each time manually.
+
+Everything until this point helped build the container image. Those commands only run once when the container image is built or rebuilt. The final `CMD` line is what runs every time the container is started.
+
+### Creating the install script
+
+The alpine linux distribution is slim and doesn't include bash, but we can access a shell at `/bin/sh`. Into the `scripts` folder at the root of your project create the following file:
```

Comment:
a shell -> the "busybox" shell, which is compatible with the basics

Reply:
Altered.